### PR TITLE
Add clearlinux dockerfile

### PR DESCRIPTION
**Pull request by jascott1**
Adds a Dockerfile for Clear Linux

### CODE REVIEW

File: demo/openssl-qat-engine/Dockerfile.clear [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    tar zxf $QAT_DRIVER_RELEASE.tar.gz
+
+RUN sed -i -e 's/cmn_ko$//' -e 's/lac_kernel$//' quickassist/Makefile && \
+    KERNEL_SOURCE_ROOT=/tmp ./configure && \
+    make quickassist-all adf-ctl-all && \
+    install -m 755 build/libqat_s.so /usr/lib/ && \
+    install -m 755 build/libusdm_drv_s.so /usr/lib/ && \
+    install -m 755 build/adf_ctl /usr/bin/ && \
+    cd openssl && ./config && cd - && \
+    mkdir -p /usr/lib/x86_64-linux-gnu/engines-1.1 && \
```
[the remark this answers is not in the document]
Comment (jascott1) [reply demo/openssl-qat-engine/Dockerfile.clear]: Fixed!

**Comment by jascott1** [comment]
@mythi Can this be merged?

File: demo/openssl-qat-engine/Dockerfile.clear [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    --with-qat_install_dir=/usr/lib \
+    --enable-openssl_install_build_arch_path && \
+    make && make install
+
+FROM clearlinux:base
+
+RUN swupd bundle-add openssl
+COPY --from=builder /usr/lib/libqat_s.so /usr/lib/
+COPY --from=builder /usr/lib/libusdm_drv_s.so /usr/lib/
+COPY --from=builder /usr/bin/adf_ctl /usr/bin
```
Comment (mythi) [inline demo/openssl-qat-engine/Dockerfile.clear]:
```
sh-5.0# adf_ctl
adf_ctl: error while loading shared libraries: libstdc++.so.6: cannot open shared object file: No such file or directory
```

That comes from ```c-basic``` bundle. Copy that file along with ```adf_ctl```?

File: demo/openssl-qat-engine/Dockerfile.clear [hunk 3]
```diff
@@ -0,0 +1,40 @@
+FROM clearlinux:base as builder
+
+ENV QAT_DRIVER_RELEASE="qat1.7.l.4.3.0-00033"
+ENV QAT_ENGINE_VERSION="v0.5.39"
+ENV PERL5LIB="/openssl"
+
+# add trusted CAs
+RUN clrtrust generate
+RUN swupd bundle-add --skip-diskspace-check os-core-dev c-basic network-basic
```
[the remark this answers is not in the document]
Comment (mythi) [reply demo/openssl-qat-engine/Dockerfile.clear]: It's better to combine these ```RUN``` stages with ```&&```

[the remark this answers is not in the document]
Comment (mythi) [reply demo/openssl-qat-engine/Dockerfile.clear]: Same with ```git```.

**Comment by mythi** [comment]
@jascott1 one more. Need to copy the licenses in the image.

File: demo/openssl-qat-engine/Dockerfile.clear (hunk 3 above)
[the remark this answers is not in the document]
Comment (mythi) [reply demo/openssl-qat-engine/Dockerfile.clear]: Edit2: would ```devpkg-systemd``` be enough to replace ```os-core-dev```

[the remark this answers is not in the document]
Comment (jascott1) [reply demo/openssl-qat-engine/Dockerfile.clear]: All of above fixed. Also had to add `devpkg-openssl`.

File: demo/openssl-qat-engine/Dockerfile.clear [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    --with-openssl_install_dir=/usr/lib64 \
+    --enable-upstream_driver \
+    --enable-usdm \
+    --with-qat_install_dir=/usr/lib \
+    --enable-openssl_install_build_arch_path && \
+    make && make install
+
+FROM clearlinux:base
+
+RUN swupd bundle-add openssl
```
[the remark this answers is not in the document]
Comment (jascott1) [reply demo/openssl-qat-engine/Dockerfile.clear]: IIRC thats an intermittent error on the backend. I did not see this when working with and updating the Dockerfile.clear

File: demo/openssl-qat-engine/Dockerfile.clear [hunk 5]
```diff
@@ -0,0 +1,40 @@
+FROM clearlinux:base as builder
+
+ENV QAT_DRIVER_RELEASE="qat1.7.l.4.3.0-00033"
+ENV QAT_ENGINE_VERSION="v0.5.39"
```
[the remark this answers is not in the document]
Comment (jascott1) [reply demo/openssl-qat-engine/Dockerfile.clear]: Fixed!

File: demo/openssl-qat-engine/Dockerfile.clear (hunk 2 above)
Comment (jascott1) [reply demo/openssl-qat-engine/Dockerfile.clear]: After latest changes I see the following:
```
root@ef7bf203f20f / # adf_ctl
Starting all devices.
Can not open /dev/qat_adf_ctl
```

**Comment by jascott1** [comment]
@mythi I copied the LICENSE.GPL file from the builder image to final image.

File: demo/openssl-qat-engine/Dockerfile.clear (hunk 4 above)
[the remark this answers is not in the document]
Comment (mythi) [reply demo/openssl-qat-engine/Dockerfile.clear]: It does not look severe error so I'm OK.

**Comment by mythi** [comment]
@jascott1 also please rebase to latest master

**Comment by jascott1** [comment]
@mythi I added the licenses and rebased.

**Pull request merged**